rs/node_rewards/rewards_calculation/src/performance_based_algorithm/v1/e2e_tests.rs

```diff
@@ -1,40 +1,102 @@
-use crate::performance_based_algorithm::DataProvider;
+use crate::performance_based_algorithm::PerformanceBasedAlgorithmInputProvider;
+use crate::performance_based_algorithm::results::{DailyNodeFailureRate, DailyNodeRewards};
 use crate::performance_based_algorithm::test_utils::{
     test_node_id, test_provider_id, test_subnet_id,
 };
 use crate::performance_based_algorithm::v1::RewardsCalculationV1;
-use crate::types::{DayUtc, NodeMetricsDailyRaw, RewardableNode};
-use ic_base_types::{PrincipalId, SubnetId};
+use crate::types::{NodeMetricsDailyRaw, RewardableNode};
+use chrono::NaiveDate;
+use ic_base_types::{NodeId, PrincipalId, SubnetId};
 use ic_protobuf::registry::node::v1::NodeRewardType;
 use ic_protobuf::registry::node_rewards::v2::{NodeRewardRate, NodeRewardRates, NodeRewardsTable};
+use lazy_static::lazy_static;
 use maplit::btreemap;
 use rust_decimal_macros::dec;
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, HashMap};
 
-// ================================================================================================
-// Mock DataProvider
-// ================================================================================================
+// Predefined rewards table used in tests
+// Note: Values for 'xdr_permyriad_per_node_per_month' have been chosen to be give good daily
+// rewards once divided by REWARDS_TABLE_DAYS.
+lazy_static! {
+  static ref NODE_REWARDS_TABLE: NodeRewardsTable = {
+    let mut table = BTreeMap::new();
```

Comment:
Can you use 4 space indent? I guess rustfmt does not work inside of macro calls.